flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/batch/sql/FunctionITCase.java

```diff
@@ -159,6 +159,22 @@ public void testUsingAddJar() throws Exception {
                 "drop function lowerUdf");
     }
 
+    @Test
+    public void testExpressionReducerByUsingJar() {
```

Comment:
We should avoid code deduplication in the code base. Most paths around functions are identical for both batch and streaming. So I don't think we need dedicated tests for `batch/sql/FunctionITCase`. What do you think about removing `batch/sql/FunctionITCase` entirely (my recommendation) or at have just one basic feature test in there?

Reply:
Yes, I agree with you that we should avoid code deduplication, but I prefer the latter, we should have one test to prove it works in batch mode. BTW, maybe there exists some batch exclusive test in the future.